services/QuillLMS/app/controllers/teachers/classrooms_controller.rb

```diff
@@ -213,26 +230,31 @@ def transfer_ownership
   end
 
   private def format_students_for_classroom(classroom)
-    students = classroom.students
-      .ordered_by_last_name
-      .map(&:attributes)
-
-    if classroom.visible
-      classroom_unit_ids = ClassroomUnit.where(classroom:).ids
-
-      students = students.map do |student|
-        student.merge(
-          number_of_completed_activities: ActivitySession.where(state: ActivitySession::STATE_FINISHED, user_id: student['id'], classroom_unit_id: classroom_unit_ids).count || 0,
-          provider: User.find(student['id']).provider
-        )
-      end
-    end
+    # do sorting in ruby so it uses the eager loaded students
+    students = classroom
+      .students
+      .sort_by { |s| s.name.split[1] || s.name }
```

Comment:
Just flagging that it's possible we'd get some unexpected behavior here with two-part last names -- probably not a big deal in practice.